api/resource/resource.go

```diff
@@ -458,10 +403,36 @@ func (r *Resource) GetNamespace() string {
 
 // OrgId returns the original, immutable ResId for the resource.
 // This doesn't have to be unique in a ResMap.
-// TODO: compute this once and save it in the resource.
 func (r *Resource) OrgId() resid.ResId {
-	return resid.NewResIdWithNamespace(
-		r.GetGvk(), r.GetOriginalName(), r.GetOriginalNs())
+	ids := r.PrevIds()
+	if len(ids) > 0 {
+		return ids[0]
+	}
+	return r.CurId()
+}
+
+// PrevIds returns a list of ResIds that includes every
+// previous ResId the resource has had through all of its
+// GVKN transformations, in the order that it had that ID.
+// The returned array does not include the resource's current
+// ID.
+// If there are no previous IDs, this will return nil.
+func (r *Resource) PrevIds() []resid.ResId {
+	var ids []resid.ResId
+
+	// names and ns should always be the same length
```

Comment:
Replace this comment with
```
if condition {
  panic(err)
}
```
better to panic with a useful error message than an index out of range.

TODO consider merging previous names and previous namespaces into one list of pairs on one annotation instead of two, so there'd be no chance of error.

Reply:
added the if statement and a TODO comment